pilot/pkg/model/extensions.go

```diff
@@ -159,13 +159,21 @@ func (p *WasmPluginWrapper) buildPluginConfig() *wasmextensions.PluginConfig {
 
 	datasource := buildDataSource(u, plugin)
 	resourceName := p.Namespace + "." + p.Name
-	return &wasmextensions.PluginConfig{
+	ret := wasmextensions.PluginConfig{
 		Name:          resourceName,
 		RootId:        plugin.PluginName,
 		Configuration: cfg,
 		Vm:            buildVMConfig(datasource, p.ResourceVersion, plugin),
-		FailOpen:      plugin.FailStrategy == extensions.FailStrategy_FAIL_OPEN,
 	}
+
+	switch plugin.GetFailStrategy() {
```

Comment:
That's a good point. I should just add a nolint instead.

Reply:
I'm not really sure, if it's easy to check proxy version with `VersionGreaterAndEqual` here.

Reply:
Yeah, regardless id rather do this in another PR.

Reply:
I don't think this would work for 1.24. Although it uses a recent enough version of envoy, the protos shipped with 1.24 dont have these new changes fields.

Reply:
yeah, should be `1.25+`.